Network-Security/Integration/aws-python-amazon-inspector-integration/c1ns-inspector-findings.yml

```diff
@@ -0,0 +1,957 @@
+                    unprotected_cves_html += f"<th>{col}</th>" + "\n"
+                unprotected_cves_html += "</tr>" + "\n"
+                # write rows
+                for row in csv_rows_unprotected_cves:
+                    unprotected_cves_html += "<tr>" + "\n"
+                    for col in row:
+                        unprotected_cves_html += f"<td>{col}</td>" + "\n"
+                    unprotected_cves_html += "</tr>" + "\n"
+                unprotected_cves_html += "</table>"
+
+                tmp_csv_file_found_policies = TemporaryFile(mode="w+", newline="")
+                writer = csv.writer(tmp_csv_file_found_policies)
+                csv_header_found_policies = ["instance_id", "instance name", "cve", "Intrusion Prevention Filtering"]
+                writer.writerow(csv_header_found_policies)
+                for row in csv_rows_found_policies:
+                    writer.writerow(row)
+                tmp_csv_file_found_policies.seek(0)
+
+                found_policies_html = '<table border="1">' + "\n"
+                # write headers
+                found_policies_html += "</tr>" + "\n"
+                for col in csv_header_found_policies:
+                    found_policies_html += f"<th>{col}</th>" + "\n"
+                found_policies_html += "</tr>" + "\n"
+                # write rows
+                for row in csv_rows_found_policies:
+                    found_policies_html += "<tr>" + "\n"
+                    for col in row:
+                        found_policies_html += f"<td>{col}</td>" + "\n"
+                    found_policies_html += "</tr>" + "\n"
+                found_policies_html += "</table>"
+
+                # The HTML body of the email.
+                body_html = f"""<html>
+                <head></head>
+                <body>
+                    <h1>Found Policies for CVE</h1>
+                    {found_policies_html}
+                    <h1>Unprotected CVEs</h1>
+                    {unprotected_cves_html}
+                </body>
+                </html>
+                """
+
+                send_email(
+                    SENDER,
+                    RECIPIENTS,
+                    subject,
+                    body_html,
+                    [
+                        {
+                            "filename": "unprotected_instances.csv",
+                            "attachment": tmp_csv_file_unprotected_cves,
+                        },
+                        {
+                            "filename": "protected_instances.csv",
+                            "attachment": tmp_csv_file_found_policies,
+                        },
+                    ],
+                )
+
+
+            ######################################## Uncomment the following if you running on your local machine
+            lambda_handler(None, None)
+
+      FunctionName: "FindingsReportLambda"
+      Handler: "index.lambda_handler"
+      Runtime: python3.9
+      Timeout: 600
```

Comment:
Have you tested this, do we really need 10m of timeout?

Reply:
I did, it takes around 3 to 4min I put enough time out just in case in different environment where resources may be more and it may need more time out